src/Index.php

```diff
@@ -402,9 +402,10 @@ public function create(array $args = [], $options = null): Response
             throw new \TypeError(\sprintf('Argument 2 passed to "%s()" must be of type array|bool|null, %s given.', __METHOD__, \is_object($options) ? \get_class($options) : \gettype($options)));
         }
 
-        $invalidOptions = \array_diff(\array_keys($options), $allowedOptions = [
+        $endpoint = new Create();
+        $invalidOptions = \array_diff(\array_keys($options), $allowedOptions = \array_merge($endpoint->getParamWhitelist(), [
```

Comment:
If I understand this correctly, it adds an additional call to Elasticsearch each time before an index is created? Assuming index creation is not happening too often, this is probably not a problem but worth mentioning.

Reply:
No extra call to Elasticsearch, just the valid options are retrieved from the underlying endpoint class.

Reply:
Ah, good to know. I somehow assumed it gets it from ES each time.